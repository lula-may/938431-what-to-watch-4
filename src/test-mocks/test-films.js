const movieTitles = [
  `Gone With The Wind`,
  `The Godfather`,
  `Casablanca`,
  `A Streetcar Named Desire`,
  `Some Likes It Hot`,
];

const movies = movieTitles.map((title, i) => {
  return {
    title,
    id: i,
    src: `poster-${i}.jpg`
  };
});

const headerMovie = {
  title: `Jurassic Park`,
  genre: `Sci-Fi`,
  releaseYear: 1993,
  bg: `background.jpg`,
  poster: `header-poster.jpg`
};

export {movies, headerMovie};

