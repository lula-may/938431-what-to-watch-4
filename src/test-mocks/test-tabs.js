const tabs = [
  {
    id: `aa`,
    type: `First`
  },
  {
    id: `bb`,
    type: `Second`,
  },
  {
    id: `cc`,
    type: `Third`,
  }
];

export {tabs};
