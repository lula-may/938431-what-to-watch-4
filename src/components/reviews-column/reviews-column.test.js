import React from "react";
import renderer from "react-test-renderer";
import ReviewsColumn from "./reviews-column.jsx";
import {testMovies} from "../../test-mocks/test-films";

const reviews = testMovies[0][`reviews`];

it(`should render correctly ReviewsColumnComponent`, () => {
  const tree = renderer.create(
      <ReviewsColumn
        reviews={reviews}
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
