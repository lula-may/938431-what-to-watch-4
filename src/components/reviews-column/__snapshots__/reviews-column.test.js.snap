// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly ReviewsColumnComponent 1`] = `
<div
  className="movie-card__reviews-col"
>
  <div
    className="review"
  >
    <blockquote
      className="review__quote"
    >
      <p
        className="review__text"
      >
        Betty Boughter bought some butter.
      </p>
      <footer
        className="review__details"
      >
        <cite
          className="review__author"
        >
          Betty Boughter
        </cite>
        <time
          className="review__date"
          dateTime="2019-07-14"
        >
          July 15, 2019
        </time>
      </footer>
    </blockquote>
    <div
      className="review__rating"
    >
      6.6
    </div>
  </div>
  <div
    className="review"
  >
    <blockquote
      className="review__quote"
    >
      <p
        className="review__text"
      >
        Jack and Jill went up the hill.
      </p>
      <footer
        className="review__details"
      >
        <cite
          className="review__author"
        >
          Jack Daniels
        </cite>
        <time
          className="review__date"
          dateTime="2020-06-01"
        >
          June 2, 2020
        </time>
      </footer>
    </blockquote>
    <div
      className="review__rating"
    >
      7.7
    </div>
  </div>
  <div
    className="review"
  >
    <blockquote
      className="review__quote"
    >
      <p
        className="review__text"
      >
        Should old acquaintance be forgot and never brought to mind.
      </p>
      <footer
        className="review__details"
      >
        <cite
          className="review__author"
        >
          Robert Burns
        </cite>
        <time
          className="review__date"
          dateTime="2020-03-01"
        >
          March 2, 2020
        </time>
      </footer>
    </blockquote>
    <div
      className="review__rating"
    >
      8.8
    </div>
  </div>
</div>
`;
